[build] consider artifact download retries to help harden kokoro dev builds

Example failure:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':buildSearchableOptions'.
> Could not resolve all dependencies for configuration ':intellijPlatformDependency'.
   > Could not resolve com.google.android.studio:android-studio:2025.3.4.6.
     Required by:
         root project 'flutter-intellij-kokoro'
      > Could not resolve com.google.android.studio:android-studio:2025.3.4.6.
         > Could not get resource 'https://cache-redirector.jetbrains.com/intellij-dependencies/com/google/android/studio/android-studio/2025.3.4.6/android-studio-2025.3.4.6.pom'.
            > Could not GET 'https://artifacts-caching-proxy.aws.intellij.net/intellij-dependencies/com/google/android/studio/android-studio/2025.3.4.6/android-studio-2025.3.4.6.pom'. Received status code 503 from server: Service Unavailable
```

--- a/tool/kokoro/build.sh
+++ b/tool/kokoro/build.sh
@@ -13,8 +13,7 @@ ATTEMPT=1
 while [[ $ATTEMPT -le $MAX_RETRIES ]]; do
   echo "Running buildPlugin (Attempt $ATTEMPT of $MAX_RETRIES)..."
   
-  ./gradlew buildPlugin
-  exit_code=$?
+  exit_code=0\n  ./gradlew buildPlugin || exit_code=$?
   if [[ $exit_code -eq 0 ]]; then
     echo "Gradle build completed successfully."
     break

--- a/tool/kokoro/build.sh
+++ b/tool/kokoro/build.sh
@@ -1,10 +1,35 @@
 #!/bin/bash
-
 source ./tool/kokoro/setup.sh
 setup
 
 echo "kokoro build start"
 
-./gradlew buildPlugin
+MAX_RETRIES=3
+RETRY_DELAY_SECS=15
+ATTEMPT=1
+
+# Retry the build a few times to address transient failures.
+# See: https://github.com/flutter/flutter-intellij/issues/9009
+while [[ $ATTEMPT -le $MAX_RETRIES ]]; do
+  echo "Running buildPlugin (Attempt $ATTEMPT of $MAX_RETRIES)..."
+  
+  ./gradlew buildPlugin
+  exit_code=$?
+  if [[ $exit_code -eq 0 ]]; then
+    echo "Gradle build completed successfully."
+    break
+  fi
+
+  echo "Build failed on attempt $ATTEMPT."
+  
+  if [[ $ATTEMPT -eq $MAX_RETRIES ]]; then
+    echo "All $MAX_RETRIES attempts failed. Exiting."
+    exit $exit_code
+  fi
+
+  echo "Waiting $RETRY_DELAY_SECS seconds before retrying..."
+  sleep $RETRY_DELAY_SECS
+  ATTEMPT=$((ATTEMPT + 1))
+done
 
 echo "kokoro build finished"

